### DISCUSSION TITLE
Tabs in a single scrolling strip layout

### DISCUSSION THREAD

**Discussion by fbck** [Chatterino/chatterino2#5031] [Ideas]
Tabs are neat, but having a lot of them makes the interface really cramped when chatterino window is kept small (like, the size of twitch chat's width). Most of the problem with having many tabs really comes from those tabs stacking on top of each other without limit, resulting in several strips of tabs all on top of each other, like a brick wall. (i counted 10 on my window lol)
Constraining tabs to always be on a single strip would help making the tabs more compact, and giving more window space to chat.
Tabs could be on a single strip, but still be scrollable through, and it'd still have that better visibility of tabs compared to 'hiding all tabs' option, to give you an idea of where you're at in your tab list. 
It could look something like this, like the scrolling tab bars in browsers (with buttons and wider tabs that could still have full names visible)
https://i.stack.imgur.com/Kxv7K.png

**Comment by Nerixyz** [comment]
Linking to https://github.com/Chatterino/chatterino2/discussions/5030 for completeness (and because GitHub doesn't show that in discussions for whatever reason).

**Comment by fbck** [comment]
if a tab strip is scrollable, it's not as inconvenient (for example, firefox has tabs that collapse in that way and can be scrolled through, and it seems to be working pretty well there). and it could still be a better tradeoff for how obtrusive a large number of tabs can get (which is really just more than 10, about two strips, maybe three). 
it's somewhere between 'having no tabs visible at all' and 'all tabs visible', tucking most of them away, but still showing enough to have an idea about where things are in the tab list and keeping ability to navigate that list even in that compact interface. 
a dropdown list would be a sufficient and simpler solution though. (it'd just have less of immediate visibility of the list, making users click and then search for the tab they need, kinda flipping how tabs are used, where it's looking/searching and then clicking)
thanks for considering this anyway
